doc/main/overview.rst

```diff
@@ -186,8 +198,49 @@ and querying the geometry state.
 Stepping loop
 =============
 
-The stepping loop in Celeritas is a sorted loop over "actions", each of which
-is usually a kernel launch (or an inner loop over tracks if running on CPU).
+The core algorithm in Celeritas is to perform a *loop interchange*
+:cite:`allen_automatic_1984` between particle tracks and steps. The classical
+(serial) way of simulating an event is to have an outer loop over tracks and an
+inner loop over steps, and inside each step are the various actions applied to
+a track such as evaluating cross sections, calculating the distance to the
+nearest geometry boundary, and undergoing an interaction to produce
+secondaries. In Python pseudocode this looks like:
+
+.. code-block:: python
+
+   track_queue = primaries
+   while track_queue:
+      track = track_queue.pop()
+      while track.alive:
+         for apply_action in [pre, along, post]:
+            apply_action(track)
+         track_queue += track.secondaries
+
+There is effectively a data dependency between the track at step *i* and step
+*i + 1* that prevents vectorization. The approach Celeritas takes to
+"vectorize" the stepping loop on GPU is to have an outer loop over "step
+iterations" and an inner loop over "track slots":
```

Comment:
Just double checking: do we have a definition of a track slot in the documentation?

Reply:
Added. Thanks!